test/function_compilation.jl

```diff
@@ -187,4 +189,38 @@ using DataFramesMeta
         end
     end
 end
+
+@testset "composed compilation" begin
+    @eval begin
```

Comment:
same here - I do not understand the benefit of `@eval`?

Reply:
As mentioned above, `@eval` is necessary because we need to evaluate in global scope to prevent caching the anonymous function in the method table.